libqtile/widget/yahoo_weather.py

```diff
@@ -84,60 +156,78 @@ class YahooWeather(GenPollUrl):
         ('steady', 's', 'symbol for steady atmospheric pressure'),
     ]
 
-    json = False
-
     def __init__(self, **config):
-        GenPollUrl.__init__(self, **config)
+        base.ThreadedPollText.__init__(self, **config)
```

Comment:
Is there a reason to make this switch? It looks like you're manually making the request using the requests module, which is a dependency we currently do not require.

Reply:
That's a good point; I really only made the switch because at the time this was easier than setting up the url for GenPollUrl. I'll have another look.

Reply:
I am having difficulties setting up the url; the example-code on the Yahoo site is for Python 2 and doesn't work anymore - even if it did work, though, it would require another ten imports.
I'll give this some more thought tomorrow.